src/aws-cpp-sdk-core/source/utils/HashingUtils.cpp

```diff
@@ -91,16 +90,17 @@ static ByteBuffer TreeHashFinalCompute(Aws::List<ByteBuffer>& input)
 
 ByteBuffer HashingUtils::CalculateSHA256TreeHash(const Aws::String& str)
 {
-    Sha256 hash;
     if (str.size() == 0)
     {
+        Sha256 hash;
```

Comment:
do we even need object for this anymore? can we just make it a static helper?

Reply:
so in theory yes, we could make a static helper function but it would require a major re-write of of how hashing works. specifically in Sha256 we separate implementation using pimpl using a [unique ptr to a Hash type](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/utils/crypto/Sha256.h#L69). because of this pimpl indirection we dont know the underlying type and invoke it using its virtual function. We could of course remove this pimpl indirection but the delta of that change would prove to be much more than and just declaring a temp hash var and using the vtable. IMO it would create a diff that would make this change more risky than it already is.

Reply:
we could just have a static method in Sha256 that creates an instance and calls hash on it, instead of doing it all over the place? or have oneshot variant that resets state to working after hash? but whatever, not something to lose sleep over.